eng/restore/harvestPackages/harvestPackages.depproj

```diff
@@ -1,47 +1,32 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project InitialTargets="AddPackageReferences" Sdk="Microsoft.NET.Sdk">
```

Comment:
Can we instead make the harvestPackages.depproj just a .targets file and sequence its logic correctly? We should use `PackageDownload` instead of `PackageReference` here as we are only interested in downloading the packages.

Reply:
> Can we instead make the harvestPackages.depproj just a .targets file and sequence its logic correctly?

I don't see much benefit of putting this logic in a targets file as we would still need to call this from an isolated project. Also, @ericstj pointed out that setting initialTartgets on a .targets file should be done with a lot of care as it may be dangerous in case somebody else imports the targets file. He was fine on having this scoped just to this depproj. 

> We should use PackageDownload instead of PackageReference here as we are only interested in downloading the packages.

How does that work? Is it basically an item that nuget will download during the Restore target and won't calculate dependencies?

Reply:
> I don't see much benefit of putting this logic in a targets file as we would still need to call this from an isolated project.

Correct, the isolated project is Tools.proj which we use to restore all our dependencies specified by arcade and by our Tools.props file. We currently other restore logic there as well: https://github.com/dotnet/corefx/blob/master/eng/Tools.props#L54. I understand if you don't want to make the change in this PR but I would definitely like to get rid of yet another depproj.

> We should use PackageDownload instead of PackageReference here as we are only interested in downloading the packages.

Correct: https://github.com/NuGet/Home/issues/7339.

Reply:
> Tools.props

I'm not sure if setting the InitialTargets on a project like Tools.proj is a good idea, specially because that one is not even checked in to this repo, so breaking changes to this logic would not be detected until ingestion, and will not be easy to diagnose. If we decide this is the best approach then that is fine, but at least for now I rather have an extra depproj where we know that the setting of initial targets is scoped to just one project that we control.

Reply:
The Tools.proj file is just a wrapper file, we don't loose control over the restore process because of it. I don't see any issue with having InitialTargets in it. cc @ericstj

Reply:
What I propose is probably out-of-scope for this PR but I think we should follow-up on this. One thing to note is that this probably requires the packaging package in arcade to be converted to an msbuild sdk so that its task is available before the restore happens.